core/src/main/java/nl/nn/adapterframework/lifecycle/WebContentServlet.java

```diff
@@ -0,0 +1,217 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.lifecycle;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.WeakHashMap;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.detect.Detector;
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.io.TikaInputStream;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.metadata.TikaMetadataKeys;
+import org.springframework.http.MediaType;
+import org.springframework.util.MimeType;
+
+import nl.nn.adapterframework.configuration.Configuration;
+import nl.nn.adapterframework.configuration.IbisContext;
+import nl.nn.adapterframework.configuration.classloaders.ClassLoaderBase;
+import nl.nn.adapterframework.http.HttpServletBase;
+import nl.nn.adapterframework.util.ClassUtils;
+import nl.nn.adapterframework.util.LogUtil;
+
+/**
+ * 
+ * @author Niels Meijer
+ */
+@IbisInitializer
+public class WebContentServlet extends HttpServletBase {
+
+	private final transient Logger log = LogUtil.getLogger(this);
+	private static final String SERVLET_PATH = "/webcontent/";
+	private static final String WELCOME_FILE = "index.html";
+	private final Map<String, MimeType> supportedMediaTypes = new HashMap<>();
+	private final Map<URL, MimeType> computedMediaTypes = new WeakHashMap<>();
+	private IbisContext ibisContext;
+	private Detector detector = null;
+
+	@Override
+	public void init() throws ServletException {
+		super.init();
+
+		try {
+			loadMediaTypes();
+		} catch (IOException e) {
+			throw new ServletException(e);
+		}
+
+		try {
+			TikaConfig tika = new TikaConfig();
+			detector = tika.getDetector();
+		} catch (TikaException | IOException e) {
+			throw new ServletException(e);
+		}
+
+		ibisContext = IbisApplicationServlet.getIbisContext(getServletContext());
+	}
+
+	private void loadMediaTypes() throws IOException {
+		URL mappingFile = ClassUtils.getResourceURL("/MediaTypeMapping.properties");
+		if(mappingFile == null) {
+			throw new IOException("unable to find mappingFile");
+		}
+
+		try(InputStream stream = mappingFile.openStream()) {
+			Properties properties = new Properties();
+			properties.load(stream);
+			for(String key : properties.stringPropertyNames()) {
+				String value = properties.getProperty(key);
+				supportedMediaTypes.put(key, MediaType.valueOf(value));
+			}
+		}
+	}
+
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```

Comment:
Omdat dit alleen maar statische resources ophaalt, lijkt het me relevant om etag en not-modified-since startup handling toe te voegen. (Of alleen een TODO daarvoor)